tests/test_utils.py

```diff
@@ -303,3 +306,437 @@ def check_multistream_classic(fnames):
                     actual, expected, decimal=4,
                     err_msg='Variable=%s, freq=%s, how=%s: '
                             'failed comparison' % (key, freq, how))
+
+def tsplit(string, delimiters):
+    '''Behaves like str.split but supports multiple delimiters.
+
+    '''
+
+    delimiters = tuple(delimiters)
+    stack = [string,]
+
+    for delimiter in delimiters:
+        for i, substring in enumerate(stack):
+            substack = substring.split(delimiter)
+            stack.pop(i)
+            for j, _substring in enumerate(substack):
+                stack.insert(i+j, _substring)
+
+    return stack
+
+def plot_science_tests(driver, testname, science_test_data_dir, result_dir, plot_dir, vic_42_dir, vic_50_dir,
+                        obs_dir, plots_to_make):
+
+    ''' makes science test figures
+
+    Parameters
+    ----------
+    driver: <str>
+        Name of Driver
+    testname: <str>
+        Name of test
+    science_test_data_dir: <str>
+        Science test data directory
+    result_dir: <str>
+        Result directory
+    plot_dir: <str>
+        Directory for output plots
+    vic_42_dir: <str>
+        Directory for VIC 4.2 archive
+    vic_50_dir <str>
+        Directory for VIC 5.0 archive
+    obs_dir <str>
+        Directory for observations archive
+    plots_to_make <Dict>
+        Keys that indicate which plots should be made
+
+    Returns
+    ----------
+    '''
+    if testname == "science_test_snotel":
+        plot_snotel_comparison(driver,
+                                testname,
+                                result_dir,
+                                plot_dir,
+                                os.path.join(science_test_data_dir, 'test_runs', vic_42_dir, 'snotel', 'results'),
+                                os.path.join(science_test_data_dir, 'test_runs', vic_50_dir, 'snotel', 'results'),
+                                os.path.join(science_test_data_dir, 'datasets', obs_dir, 'observations'),
+                                plots_to_make)
+    elif testname == "science_test_fluxnet":
+        plot_fluxnet_comparison(driver,
+                                testname,
+                                result_dir,
+                                plot_dir,
+                                os.path.join(science_test_data_dir, 'test_runs', vic_42_dir, 'ecflux', 'results'),
+                                os.path.join(science_test_data_dir, 'test_runs', vic_50_dir, 'ecflux', 'results'),
+                                os.path.join(science_test_data_dir, 'datasets', obs_dir, 'obs'),
+                                plots_to_make)
+    else:
+        print("this has not yet been implemented in the VIC 5.0 science test suite")
+
+def plot_snotel_comparison(driver, testname, result_dir, plot_dir, vic_42_dir, vic_50_dir,
```

Comment:
I ended up reviewing the ecflux function first but all my comments on breaking this out into smaller pieces, apply here as well.